### PR TITLE
Add `body` to unit tests

### PR DESCRIPTION
**Pull request by MaxDall**
The adds the `body` to parser unit tests as well as updating the test cases for that occasion.

### CODE REVIEW

**Comment by dobbersc** [comment]
The CI fails because some publishers are missing the test case for the body attribute. 

When I ran the test case generation script, I noticed some publishers may have incorrectly extracted attributes. Did you roughly check the extracted bodies?

**Comment by dobbersc** [comment]
Weirdly the order of the parser versions in the JSON for SZ and TheNation are also in a different order compared to the current commit. I think this may be related to the path globbing, which may not guarantee a deterministic list order.

**Comment by MaxDall** [comment]
> Weirdly the order of the parser versions in the JSON for SZ and TheNation are also in a different order compared to the current commit. I think this may be related to the path globbing, which may not guarantee a deterministic list order.

I don't quite get this one. If I look into the JSONs the parser versions are in the same order as in the master. Could you provide me with a bit more context? I think I'm missing your point here

**Comment by dobbersc** [comment]
> > Weirdly the order of the parser versions in the JSON for SZ and TheNation are also in a different order compared to the current commit. I think this may be related to the path globbing, which may not guarantee a deterministic list order.
> 
> I don't quite get this one. If I look into the JSONs the parser versions are in the same order as in the master branch. Could you provide me with a bit more context? I think I'm missing your point here.

It is essentially the same as with the order of attributes. I just have not pushed the changes as agreed. You can see the diff for TheNation.json here: https://www.diffchecker.com/6S7112p6/

File: src/fundus/parser/data.py [hunk 1]
```diff
@@ -231,8 +272,18 @@ class ArticleSection(TextSequenceTree):
     headline: TextSequence
```
[the remark this answers is not in the document]
Comment (MaxDall) [reply src/fundus/parser/data.py:247]: Headline was originally intended to be a TextSequence to catch any errors resulting from consequent headlines.

File: tests/utility.py [hunk 2]
```diff
@@ -88,25 +90,32 @@ class ExtractionEncoder(json.JSONEncoder):
     def default(self, obj: object):
         if isinstance(obj, datetime.datetime):
             return str(obj)
+        if isinstance(obj, TextSequenceTree):
```
[the remark this answers is not in the document]
Comment (MaxDall) [reply tests/utility.py:111]: Sounds good to me!

**Comment by MaxDall** [comment]
@dobbersc Is there anything left open here? I would like to merge this PR.

File: scripts/generate_parser_test_files.py [hunk 3]
```diff
@@ -123,11 +123,15 @@ def main() -> None:
             for html in html_mapping.values():
                 versioned_parser = html.publisher.parser(html.crawl_date)
                 extraction = versioned_parser.parse(html.content)
-                new = {attr: value for attr, value in extraction.items() if attr in arguments.attributes}
+                missing_attributes = set(arguments.attributes) - set(
+                    test_data.get(type(versioned_parser).__name__) or {}
```
Comment (addie9800) [inline scripts/generate_parser_test_files.py:127]: If I'm not mistaken, this line is only needed to update old test cases. Shouldn't it be removed as soon as all old test data is updated?

**Review by addie9800 — COMMENTED** [review]
This is a nice addition, there is one merge conflict left though

File: scripts/generate_parser_test_files.py (hunk 3 above)
Comment (MaxDall) [reply scripts/generate_parser_test_files.py:127]: Depends if we want to add additional attributes in the future ;)

Comment (addie9800) [reply scripts/generate_parser_test_files.py:127]: True :)

**Pull request merged**